dashboard/app/controllers/xhr_proxy_controller.rb

```diff
@@ -0,0 +1,47 @@
+# A controller for proxying web requests to 3rd party APIs. This protects
+# applab users from XSS attacks, complies with the Same Origin Policy,
+# and prevents http/https mismatch warnings in IE.
+#
+# Responses are cached for one minute since many 3rd party APIs serve data
+# which changes frequently such as news or sports scores.
+#
+# To reduce the likelihood of abuse, we only proxy content with an allowed
+# whitelist of media types. We will need to monitor usage to detect
+# abuse and potentially add other abuse prevention measures.
+
+require 'set'
+
+class XhrProxyController < ApplicationController
+  include ProxyHelper
+
+  ALLOWED_CONTENT_TYPES = Set.new(['application/json'])
```

Comment:
According to https://en.wikipedia.org/wiki/JSON, although applicaiton/json is the official mime types, you might also expect to see text/json and text/javascript in the wild.  I don't think there is too much harm in allowing either of these.